cartographer_ros/src/cartographer_node_main.cc

```diff
@@ -277,8 +283,13 @@ Rigid3d Node::LookupToTrackingTransformOrThrow(const carto::common::Time time,
 void Node::AddOdometry(int64 timestamp, const string& frame_id,
                        const Rigid3d& pose, const PoseCovariance& covariance) {
   const carto::common::Time time = carto::common::FromUniversal(timestamp);
+  PoseCovariance maybe_constant_variance = covariance;
+  if (options_.use_constant_odometry_variance) {
+    maybe_constant_variance =
+        PoseCovariance::Identity() * options_.constant_odometry_variance;
```

Comment:
I guess we'll eventually want separate variances for translational and rotational components.